blockchain/utxolookup.go

```diff
@@ -0,0 +1,548 @@
+	if !ok {
+		return true
+	}
+
+	return entry.outputs[packedIndex].spent
+}
+
+// SpendOutput marks the output at the provided index as spent.  Specifying an
+// output index that does not exist will not have any effect.
+func (entry *UtxoEntry) SpendOutput(outputIndex uint32) {
+	packedIndex, ok := entry.sparseOutputs[outputIndex]
+	if !ok {
+		return
+	}
+
+	// Nothing to do if the output is already spent.
+	output := &entry.outputs[packedIndex]
+	if output.spent {
+		return
+	}
+
+	entry.modified = true
+	output.spent = true
+	return
+}
+
+// IsFullySpent returns whether or not the transaction the utxo entry represents
+// is fully spent.
+func (entry *UtxoEntry) IsFullySpent() bool {
+	// The entry is not fully spent if any of the outputs are unspent.
+	for _, output := range entry.outputs {
+		if !output.spent {
+			return false
+		}
+	}
+
+	return true
+}
+
+// Amount returns the amount of the provided output index.
+//
+// Returns 0 if the output index references an output that does not exist
+// either due to it being invalid or because the output is not part of the store
+// due to previously being spent/pruned.
+func (entry *UtxoEntry) Amount(outputIndex uint32) int64 {
+	packedIndex, ok := entry.sparseOutputs[outputIndex]
+	if !ok {
+		return 0
+	}
+
+	return entry.outputs[packedIndex].amount
+}
+
+// PkScript returns the public key script for the provided output index.
+//
+// Returns nil if the output index references an output that does not exist
+// either due to it being invalid or because the output is not part of the store
+// due to previously being spent/pruned.
+func (entry *UtxoEntry) PkScript(outputIndex uint32) []byte {
+	packedIndex, ok := entry.sparseOutputs[outputIndex]
+	if !ok {
+		return nil
+	}
+
+	return entry.outputs[packedIndex].pkScript
+}
+
+// newUtxoEntry returns a new unspent transaction output entry with the provided
+// coinbase flag and block height ready to have unspent outputs added.
+func newUtxoEntry(version int32, isCoinBase bool, blockHeight int32) *UtxoEntry {
+	return &UtxoEntry{
+		version:       version,
+		isCoinBase:    isCoinBase,
+		blockHeight:   blockHeight,
+		sparseOutputs: make(map[uint32]uint32),
+	}
+}
+
+// UtxoStore provides a subset of the set of unspent transaction outputs from a
+// specific point of view in the chain.  For example, it could be from the end
+// of the main chain, some point in the history of the main chain, or down a
+// side chain.
+//
+// The unspent outputs are need by other transactions for things such as script
+// validation and double spend prevention.
+type UtxoStore map[wire.ShaHash]*UtxoEntry
+
+// Entry returns information about a given transaction according to the current
+// state of the store.  It will return nil if the passed transaction hash does
+// not exist in the store or is otherwise not available such as when it has been
+// disconnected during a reorg.
+func (store UtxoStore) Entry(txHash *wire.ShaHash) *UtxoEntry {
```

Comment:
I agree a better name could be beneficial here.  I don't either `GetEntry` or `FetchEntry` would work well though because `GetEntry` is against [effective Go guidelines](http://golang.org/doc/effective_go.html) and in general `FetchX` is used throughout the code base to mean it's actually hitting the database such as in the case of `FetchUtxoStore`, `fetchInputUtxos`, etc.

In this case, it's simply attempting to return a transaction entry from the utxo store which may may not be there (a map lookup).  How about `LookupEntry`?